### PR TITLE
Debugger stop reason

### PR DESCRIPTION
**Pull request by Michaelangel007**
Small QoL when a breakpoint is triggered.  See bottom line in debugger.  Color code register and breakpoint number.

# Before
debugger_stop_reason_before

# After
debugger_stop_reason_after

### CODE REVIEW

File: source/Debugger/Debug.cpp [hunk 1]
```diff
@@ -85,7 +85,8 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 	static DebugBreakOnDMA g_DebugBreakOnDMA[NUM_BREAK_ON_DMA];
 	static DebugBreakOnDMA g_DebugBreakOnDMAIO;
 
-	int  g_bDebugBreakpointHit = 0;	// See: BreakpointHit_t
+	int                 g_bDebugBreakpointHit = 0;       // See: BreakpointHit_t
+	static Breakpoint_t*g_pDebugBreakpointHit = nullptr; // NOTE: Only valid for BP_HIT_REG, see: CheckBreakpointsReg()
```
Comment (tomcw) [inline source/Debugger/Debug.cpp:89]: Minor: probably could do with a space between the type & the variable.

File: source/Debugger/Debug.cpp [hunk 2]
```diff
@@ -8604,7 +8608,22 @@ void DebugContinueStepping(const bool bCallerWillUpdateDisplay/*=false*/)
 			else if (g_bDebugBreakpointHit & BP_HIT_OPCODE)
 				stopReason = "Opcode match";
 			else if (g_bDebugBreakpointHit & BP_HIT_REG)
-				stopReason = "Register matches value";
+			{
+					if (g_pDebugBreakpointHit)
+					{
+						int iBreakpoint = (g_pDebugBreakpointHit - g_aBreakpoints);
+						stopReason = StrFormat( "Register %s%s%s matches breakpoint %s#%s%d",
+							CHC_CATEGORY, // TODO: CHC_REGS
```
Comment (tomcw) [inline source/Debugger/Debug.cpp:8616]: Ideally fix this TODO.

**Review by tomcw — APPROVED** [review]

File: source/Debugger/Debug.cpp (hunk 1 above)
Comment (Michaelangel007) [reply source/Debugger/Debug.cpp]: Thanks for catching that!

File: source/Debugger/Debug.cpp (hunk 2 above)
Comment (Michaelangel007) [reply source/Debugger/Debug.cpp]: Thanks for the nudge!  Fixed.

**Comment by tomcw** [comment]
Tweak the HDD DMA stop reason msg to match this PR (79024f0)
NB. HDD DMA can trigger on multiple BPs...

BP-reason-for-DMA

**Comment by Michaelangel007** [comment]
@tomcw I've cleaned up a few more messages with commit a68a29bc.  Could you add a quick repro. strep for the HDD DMA breakpoints? Thanks.

**Comment by tomcw** [comment]
Sure. For the HDD DMA, use this test from the "AppleWin-Test" repo:
https://github.com/AppleWin/AppleWin-Test/tree/master/v2/HDD-RW-IO-Boundary

1. Clone the "AppleWin-Test" (if you don't already have it)
2. Launch AppleWin, F12 to load this state file: "HDD-RW-IO-Boundary.aws.yaml"
3. F7 (enter debugger), then paste in the contents of "DebuggerAutoRun.txt"
4. F7 (exit debugger), and AppleWin will run the test code, hit the next breakpoint and enter the debugger
5. Keep pressing F7 for all the breakpoint tests

NB. Once complete, from the monitor `6000G` to repeat the tests.

**Comment by tomcw** [comment]
I've checked this and it looks fine (and much better!) - thanks.

**Pull request merged**